test/manual_wasm_instantiate.html

```diff
@@ -172,6 +172,11 @@
         wasm.then(function(wasmBinary) {
           console.log('wasm download finished, begin instantiating');
           var wasmInstantiate = WebAssembly.instantiate(new Uint8Array(wasmBinary), imports).then(function(output) {
+            // When overriding instantiateWasm, in asan builds, we also need
+            // to take case of creating the WasmOffsetConverter
+            if (typeof WasmOffsetConverter != "undefined") {
```

Comment:
Can this not be done automatically for the user? This seems like a tricky thing for a user to figure out that they need to do.

Reply:
Yeah, it seems like a bug, at least in our documentation for `instantiateWasm`.     I don't think there is any easy to way to automate it though, at least not without changing the API of `instantiateWasm`.  The problem is that you need access to the wasm bits if you want to make the offset converter work.    And when `instantiateWasm` is overriden the emscripten code doesn't see the wasm bits (IIUC).

Reply:
This issue is not created by this PR though, its always been there.

Reply:
How did the test pass before then? That's what confuses me.

Reply:
Because this line was creating a dummy/empty offset coverter:

```
wasmSourceMap = resetPrototype(WasmSourceMap, Module['wasmSourceMapData']);
```

Because of the way `resetPrototype` is implemented even though `Module['wasmSourceMapData']` is undefined we were creating a valid/empty `wasmSourceMap`

Reply:
They new behaviour is that asan will assert that it doesn't have an offset converter, which I think is better than the old behaviour of just having an empty one.

Reply:
Now I see, thanks!